lib/common/formats.js

```diff
@@ -358,6 +358,126 @@ module.exports = {
+   *           "format": "typescript/es6-declarations",
+   *           "destination": "colors.d.ts"
+   *         }
+   *       ]
+   *     }
+   *   }
+   * }
+   * ```
+   *
+   * @memberof Formats
+   * @kind member
+   * @example
+   * ```typescript
+   * export const ColorBackgroundBase : string;
+   * export const ColorBackgroundAlt : string;
+   * ```
+   */
+  'typescript/es6-declarations': function(dictionary) {
+    return fileHeader(this.options) +
+      dictionary.allProperties.map(function(prop) {
+        var to_ret_prop = 'export const ' + prop.name + ' : string;';
+        if (prop.comment)
+          to_ret_prop = to_ret_prop.concat(' // ' + prop.comment);
+        return to_ret_prop;
+      }).join('\n');
+  },
+
+  /**
+   * Creates TypeScript declarations for CommonJS module
+   *
+   * ```json
+   * {
+   *   "platforms": {
+   *     "ts": {
+   *       "transformGroup": "js",
+   *       "files": [
+   *         {
+   *           "format": "javascript/module",
+   *           "destination": "colors.js"
+   *         },
+   *         {
+   *           "format": "typescript/module-declarations",
+   *           "destination": "colors.d.ts"
+   *         }
+   *       ]
+   *     }
+   *   }
+   * }
+   * ```
+   *
+   * @memberof Formats
+   * @kind member
+   * @example
+   * ```typescript
+   * export default tokens;
+   * declare interface DesignToken { value: string; name?: string; path?: string[]; comment?: string; attributes?: any; original?: any; }
+   * declare const tokens: {
+   *  "color": {
+   *    "red": DesignToken
+   *  }
+   * }
+   * ```
+   *
+   * As you can see above example output this does not generate 100% accurate d.ts.
+   * This is a compromise between of what style-dictionary can do to help and not bloating the library with rarely used dependencies.
+   *
+   * Thankfully you can extend style-dictionary very easily:
+   *
+   * ```js
+   * const JsonToTS = require('json-to-ts');
+   * StyleDictionaryPackage.registerFormat({
+   *   name: 'typescript/accurate-module-declarations',
+   *   formatter: function(dictionary) {
+   *     return 'declare const root: RootObject\n' +
+   *     'export default root\n' +
+   *     JsonToTS(dictionary.properties).join('\n');
+   *   },
+   * });
+   * ```
+   */
+  'typescript/module-declarations': function(dictionary) {
+    function treeWalker(obj) {
+      let type = Object.create(null);
+      let has = Object.prototype.hasOwnProperty.bind(obj);
+      if (has('value')) {
+        type = 'DesignToken';
+      } else {
+        for (var k in obj) if (has(k)) {
+          switch (typeof obj[k]) {
+            case 'object':
+              type[k] = treeWalker(obj[k]);
+          }
+        }
+      }
+      return type;
+    }
+
+    var file = fileHeader(this.options) +
+      'export default tokens;\n' +
+      'declare interface DesignToken { value: string; name?: string; path?: string[]; comment?: string; attributes?: any; original?: any; }\n' +
```

Comment:
Sorry @Tiamanti, one more question for you...

The DesignToken interface here and the [Prop interface](https://github.com/amzn/style-dictionary/blob/3.0/types/index.d.ts#L63-L90) are fairly similar - minus optional fields.

Is there a way to tie these together somehow? Or combine them?

If we decide to change the structure later by adding additional metadata properties, it would be nice to only have one place to fix.

I apologize again if my TypeScript naivety is showing.

Reply:
@Tiamanti did you have thoughts here?

We will likely do another release candidate of 3.0 here shortly - and it might be the last one. Would love to get this PR in as part of that.

Thanks for all your hard work!